### PR TITLE
fix(test): Filter ES6 features from node v0.12. Fixes #3552

### LINKED ISSUE #3552: test broken in node v0.12
**Do you want to request a *feature* or report a *bug*?**
bug

**What is the current behavior?**
``` text
  1) ConfigTestCases target node-dynamic-import should compile:
     Uncaught /home/travis/build/webpack/webpack/test/js/config/target/node-dynamic-import/bundle0.js:221
			.then((result) => {result.should.be.eql(expected); callback()})
			               ^^
Unexpected token =>
  /home/travis/build/webpack/webpack/test/js/config/target/node-dynamic-import/bundle0.js:221
  			.then((result) => {result.should.be.eql(expected); callback()})
  			               ^^
  SyntaxError: Unexpected token =>
      at Object.exports.runInThisContext (vm.js:73:16)
```

**If the current behavior is a bug, please provide the steps to reproduce.**
run tests on node v0.12 or check CI

**What is the expected behavior?**
test is filtered by a `test.filter.js` file if arrow function is not supported.

### PR DESCRIPTION
**Pull request by TheLarkInn**
**What kind of change does this PR introduce?**
Bugfix

**Did you add tests for your changes?**
N/A

**If relevant, link to documentation update:**
N/A

**Summary**
Filters ES6 features so arrow functions don't fail node 0.12 builds on travis. 

**Does this PR introduce a breaking change?**
No

**Other information**
Fixes #3552

### CODE REVIEW

**Comment by sokra** [comment]
it's not fixed...

looks like `test.filter.js` is not implemented in configCases (only in cases). You need to add this feature to the test infrastructure.

**Comment by TheLarkInn** [comment]
Awesome

**Comment by TheLarkInn** [comment]
This has turned out to be much more challenging than I thought. Problem is that for me to filter and validate I need to pull the config, however, there are some configs that require that a build be run before others (IE: dll-* tests). 

@sokra Do you have any suggestions on how I could still use this pattern for filtering?

**Comment by sokra** [comment]
Just omit the config parameter in test.filter.js. in the configVases config is known by the test.

**Comment by TheLarkInn** [comment]
Awesome I'll get that done later

**Comment by TheLarkInn** [comment]
Okay I figured out the problem finally. Feel free and look it over again.

File: test/configCases/target/node-dynamic-import/test.filter.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+var supportsES6 = require("../../../helpers/supportsES6");
```
Comment (sokra) [inline test/configCases/target/node-dynamic-import/test.filter.js]: That's the wrong test. node 0.12 supports ES6 classes, but no arrow function.

So you need to add a new helper (i. e. supportsArrowFunction) to test correctly.

Comment (TheLarkInn) [inline test/configCases/target/node-dynamic-import/test.filter.js]: Gauhhhh thats right okay, I'll flip them.

**Comment by sokra** [comment]
Thanks

**Pull request merged**